src/NightwatchServiceProvider.php

```diff
@@ -111,13 +108,19 @@ public function register(): void
             $this->registerAndCaptureConfig();
             $this->registerBindings();
         } catch (Throwable $e) {
-            $this->handleUnrecoverableException($e);
+            $this->registerException = $e;
```

Comment:
Instead of calling `Nightwatch::unrecoverableExceptionOccurred($e);` directly, we capture the exception and then call `Nightwatch::unrecoverableExceptionOccurred($e);` in the `boot` method instead.

The Nightwatch service provider will run before the consuming application's service provider. This pattern allows applications to call:

```php
`Nightwatch::handleUnrecoverableExceptionsUsing($callback);` 
```

In the `register` method of their service providers.

Any exception during the `register` method of our provider will then correctly be sent to the registered callback.